src/models/base.es6.js

```diff
@@ -1,59 +1,41 @@
-import { EventEmitter } from 'events';
+const THING_ID_REGEX = new RegExp('t\\d_[0-9a-z]+', 'i');
 
 class Base {
-  constructor (props={}, opts={}) {
+  constructor (props={}) {
     this.props = {};
```

Comment:
:no_bell: I'll try to find the Google I/O talk about this, but I think we should consider using a real class here because: https://developers.google.com/v8/design?hl=en#fast-property-access. I'm happy to make up some performance tests and open a pr for it